tests/baselines/reference/verbatim-declarations-parameters.js

```diff
@@ -0,0 +1,74 @@
+//// [tests/cases/compiler/verbatim-declarations-parameters.ts] ////
+
+//// [verbatim-declarations-parameters.ts]
+type Map = {} & { [P in string]: any }
+type MapOrUndefined = Map | undefined | "dummy"
+export class Foo {
+  constructor(
+    // Type node is accurate, preserve
+    public reuseTypeNode?: Map | undefined,
+    public reuseTypeNode2?: Exclude<MapOrUndefined, "dummy">,
+    // Resolve type node, requires adding | undefined
+    public resolveType?: Map,
+  ) { }
+}
+
+export function foo1(
+    // Type node is accurate, preserve
+    reuseTypeNode: Map | undefined = {},
+    reuseTypeNode2: Exclude<MapOrUndefined, "dummy">  = {},
+    // Resolve type node, requires adding | undefined
+    resolveType: Map = {}, 
+    requiredParam: number) {
+
+}
+
+
+//// [verbatim-declarations-parameters.js]
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.foo1 = exports.Foo = void 0;
+var Foo = /** @class */ (function () {
+    function Foo(
+    // Type node is accurate, preserve
+    reuseTypeNode, reuseTypeNode2, 
+    // Resolve type node, requires adding | undefined
+    resolveType) {
+        this.reuseTypeNode = reuseTypeNode;
+        this.reuseTypeNode2 = reuseTypeNode2;
+        this.resolveType = resolveType;
+    }
+    return Foo;
+}());
+exports.Foo = Foo;
+function foo1(
+// Type node is accurate, preserve
+reuseTypeNode, reuseTypeNode2, 
+// Resolve type node, requires adding | undefined
+resolveType, requiredParam) {
+    if (reuseTypeNode === void 0) { reuseTypeNode = {}; }
+    if (reuseTypeNode2 === void 0) { reuseTypeNode2 = {}; }
+    if (resolveType === void 0) { resolveType = {}; }
+}
+exports.foo1 = foo1;
+
+
+//// [verbatim-declarations-parameters.d.ts]
+type Map = {} & {
+    [P in string]: any;
+};
+type MapOrUndefined = Map | undefined | "dummy";
+export declare class Foo {
+    reuseTypeNode?: Map | undefined;
+    reuseTypeNode2?: Exclude<MapOrUndefined, "dummy">;
+    resolveType?: {
+        [x: string]: any;
+    } | undefined;
+    constructor(reuseTypeNode?: Map | undefined, reuseTypeNode2?: Exclude<MapOrUndefined, "dummy">, resolveType?: {

```

Comment:
I assume this is preexisting, but why is the "resolveType" prop/parameter resolved at all, and not a simple rewrite like the other params? Seems odd to me that reuseTypeNode is _more_ complex and doesn't get rewritten, but resolveType does and is.

Reply:
`resolveType` requires adding `undefined` to the type. So we can't directly use the original type as it was written, unless it already contains `undefined`

Reply:
I see. How does this work in the context of isolatedDEclarations, where we presumably would instead want to syntactically do this? Isn't the intent to just add `| undefined` unconditionally?

Reply:
I was planning to do that in one of the follow on PRs since that would arguably not be an improvement to current emit.  But I could add it as part of this PR as well if you think it's uncontroversial.

Reply:
I feel moderately like they're tied together, but I suppose that this PR is okay as-is as it's really just saying "if the type already has undefined, don't add it", which is a simple enough thing to reason about.

Reply:
I guess this is still an improvement as it leads to increased node reuse...

Reply:
It will have an effect. If we switch to always add ` | undefined` it will always reuse the existing type node (now it's only reuses it if we don't need to add `undefined` because the declared type already includes it)